manifests/bosh-manifest/scripts/generate-bosh-certs.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/sh
+
+set -e
+set -u
+
+CA_NAME="bosh-CA"
+
+# List of certs to generate
+# Format:
+#
+# <name_cert>,<domain1>[,domain2,domain3,...]
+#
+# Note: ALWAYS add a comma after <name_cert>, even if there are no domains
+#
+CERTS_TO_GENERATE="
+default.nats.bosh-internal,bosh.${SYSTEM_DNS_ZONE_NAME}
+default.director.bosh-internal,
+default.hm.bosh-internal,
+bosh_director,bosh.${SYSTEM_DNS_ZONE_NAME},bosh-external.${SYSTEM_DNS_ZONE_NAME}
+"
+
+generate_cert() {
+  _cn="${1}"
+  _domains="${2}"
+  _target_dir="${3}"
+
+  certstrap request-cert \
+    --passphrase "" \
+    --common-name "${_cn}" \
+		${domains:+--domain "${_domains}"}
+  certstrap sign \
+    --CA "${CA_NAME}" \
+    --passphrase "" \
+    --years "2" \
+    "${_cn}"
+
+  mv "out/${_cn}.key" "${_target_dir}/"
+  mv "out/${_cn}.csr" "${_target_dir}/"
+  mv "out/${_cn}.crt" "${_target_dir}/"
+}
+
+rotate_cert() {
+  _cn="${1}"
+  _domains="${2}"
+  _target_dir="${3}"
+
+  mv "${_target_dir}/${_cn}.key" "${_target_dir}/${_cn}_old.key"
+  mv "${_target_dir}/${_cn}.csr" "${_target_dir}/${_cn}_old.csr"
+  mv "${_target_dir}/${_cn}.crt" "${_target_dir}/${_cn}_old.crt"
+
+  generate_cert "${_cn}" "${_domains}" "${_target_dir}"
+}
+
+CERTS_DIR=$(cd "$1" && pwd)
+CA_TARBALL="$2"
+ACTION="${3:-}"
+if [ "${ACTION}" != "create" ] && [ "${ACTION}" != "rotate" ]; then
+  cat <<EOF
+Usage:
+  $0 <create|rotate>
+EOF
+  exit 1
+fi
+
+WORKING_DIR="$(mktemp -dt generate-cf-certs.XXXXXX)"
+trap 'rm -rf "${WORKING_DIR}"' EXIT
+
+mkdir "${WORKING_DIR}/out"
+echo "Extracting ${CA_NAME} cert"
+tar -xvzf "${CA_TARBALL}" -C "${WORKING_DIR}/out"
+
+cd "${WORKING_DIR}"
+for cert_entry in ${CERTS_TO_GENERATE}; do
+  cn=${cert_entry%%,*}
+  domains=${cert_entry#*,}
+
+  if [ -f "${CERTS_DIR}/${cn}.crt" ]; then
+    echo "Certificate ${cn} is already generated."
+    if [ "${ACTION}" = "rotate" ]; then
```

Comment:
Should we have a rotate job for this pipeline? Do we have any monitoring of internal cert expiry?

Reply:
No, I did not add the rotate job. Shall we?

No, we do not have any monitoring for internal certificates, neither here or in paas-cf.

To be honest, I was thinking that maybe we need a step to test all the generated certificates, to verify expire date, CNs, SAN, etc. For instance, in this case, we can check that the CN matches the format required by nats and that the host alias is registered properly. 

On the other hand, we shall ditch all this logic soon as we move to bosh 2 interpolation.

Reply:
I'm not sure. It seems like we should have both but I don't know whether this story is the right place. @alext do you have any thoughts?

Reply:
I added the jobs to rotate those certs, but we still do not have logic to rotate the bosh-CA. We use the bosh-CA as CA for nats clients.

They are completely safe to rotate and it shouldn't meant downtime.

Reply:
We already have a story in the icebox to monitor the internal certs (https://www.pivotaltracker.com/story/show/154917736), I think this can be included in that.

It would be good to cover rotating these certs here, but I think the CA cert is out of scope as that affects CF as well as bosh.